src/rgw/rgw_rados.cc

```diff
@@ -5875,25 +5875,42 @@ int RGWRados::select_new_bucket_location(RGWUserInfo& user_info, const string& z
 
   /* now check that tag exists within zonegroup */
   /* find placement rule. Hierarchy: request rule > user default rule > zonegroup default rule */
-  string rule = request_rule;
-  if (rule.empty()) {
-    rule = user_info.default_placement;
-    if (rule.empty())
-      rule = zonegroup.default_placement;
+  std::string rule;
+
+  if (!request_rule.empty()) {
+    if (zonegroup.has_placement_target(request_rule)) {
+      rule = request_rule;
+    } else {
+      ldout(cct, 0) << "could not find requested placement id " << rule << " within zonegroup " << dendl;
+      return -EINVAL;
+    }
   }
 
-  if (rule.empty()) {
-    ldout(cct, 0) << "misconfiguration, should not have an empty placement rule name" << dendl;
-    return -EIO;
+  if (rule.empty() && !user_info.default_placement.empty()) {
+    if (zonegroup.has_placement_target(user_info.default_placement)) {
+      rule = user_info.default_placement;
+    } else {
+      ldout(cct, 0) << "could not find user default placement id " << rule << " within zonegroup " << dendl;
+      return -EINVAL;
```

Comment:
@cbodley Agree, By using `ERR_INVALID_LOCATION_CONSTRAINT ` user will get the S3 error code, which can clearly point out the error reason.  I've also add `ERR_ZONEGROUP_DEFAULT_PLACEMENT_MISCONFIGURATION` to instead `EIO`. what do you think?